src/manifest.rs

```diff
@@ -37,13 +37,15 @@ impl Asset {
             chmod,
         }
     }
-    pub fn is_binary_executable(&self, release_dir_prefix: &Path) -> bool {
-        self.source_file.starts_with(release_dir_prefix) && 0 != (self.chmod & 0o111)
+    pub fn is_binary_executable(&self, workspace_root: &Path, release_dir_prefix: &Path) -> bool {
+        let source_file_abspath = workspace_root.join(&self.source_file);
```

Comment:
This still needs to rewrite `target/release/xxx` to `target/<triple>release/xxx` I think?